nominal/core/_clientsbunch.py

```diff
@@ -40,6 +56,42 @@ def write_nominal_batches(self, auth_header: str, data_source_rid: str, request:
         _path = f"/storage/writer/v1/nominal/{data_source_rid}"
         self._request("POST", self._uri + _path, params={}, headers=_headers, data=request)
 
+    def write_nominal_batches_with_metrics(
+        self,
+        auth_header: str,
+        data_source_rid: str,
+        request: bytes,
+        oldest_timestamp: IntegralNanosecondsUTC,
+        newest_timestamp: IntegralNanosecondsUTC,
+    ) -> RequestMetrics:
+        _headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/x-protobuf",
+            "Authorization": auth_header,
+        }
+        _path = f"/storage/writer/v1/nominal/{data_source_rid}"
+
+        current_time_ns = time.time_ns()
+        oldest_timestamp_diff_in_batch_before_request = (current_time_ns - oldest_timestamp) / 1e9
+        newest_timestamp_diff_in_batch_before_request = (current_time_ns - newest_timestamp) / 1e9
+
+        before_req = time.time_ns()
+        self._request("POST", self._uri + _path, params={}, headers=_headers, data=request)
+
+        current_time_ns = time.time_ns()
+        request_rtt = (current_time_ns - before_req) / 1e9
+
+        largest_e2e_rtt = (current_time_ns - oldest_timestamp) / 1e9
+        smallest_e2e_rtt = (current_time_ns - newest_timestamp) / 1e9
+
+        return RequestMetrics(
+            oldest_timestamp_diff_before_request=oldest_timestamp_diff_in_batch_before_request,
+            newest_timestamp_diff_before_request=newest_timestamp_diff_in_batch_before_request,
+            request_rtt=request_rtt,
+            largest_e2e_latency=largest_e2e_rtt,
+            smallest_e2e_latency=smallest_e2e_rtt,
+        )
```

Comment:
this is a _lot_ of lines of code for what it does - this should be cleaned up. for e.g.

```python
before_req = time.time_ns()
self._request("POST", self._uri + _path, params={}, headers=_headers, data=request)
after_response = time.time_ns()

return RequestMetrics.from_nanos(
    oldest_timestamp_diff_before_request=(before_req - oldest_timestamp),
    newest_timestamp_diff_before_request=(before_req - newest_timestamp),
    request_rtt=(after_response - before_req),
    largest_e2e_latency=(after_response - oldest_timestamp),
    smallest_e2e_latency=(after_response - newest_timestamp),
)
```